src/allmydata/storage/server.py

```diff
@@ -135,6 +136,12 @@ def __init__(self, storedir, nodeid, reserved_space=0,
         # Canaries and disconnect markers for BucketWriters created via Foolscap:
         self._bucket_writer_disconnect_markers = {}  # type: Dict[BucketWriter,(IRemoteReference, object)]
 
+    def stopService(self):
+        # Cancel any in-progress uploads:
+        for bw in list(self._bucket_writers.values()):
```

Comment:
probably don't need the `list()`?

Reply:
You do, when it gets disconnected it removes itself from `_bucket_writers` and then you're iterating over it as it gets mutated.